File: back/src/main/java/crappyUnivLife/timeMachineLetter/domain/Letter.java
package crappyUnivLife.timeMachineLetter.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@ToString
public class Letter {
    @Id @GeneratedValue
    @Column(name = "letter_id")
    private Long id;

    private String hash;

    private Boolean readable = true;
    private Boolean isEncrypted = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    @JsonIgnore
    private Member member;

    private String password;

    private Date openAt;

    private String title;

    private String content;

    private String letterFrom;

    private String letterTo;

    private Date createAt = new Date();

    // 연관관계 편의 매서드 //
    public void setMember(Member member) {
        this.member = member;
        member.getLetter().add(this);
    }
}

File: front/src/__mocks__/Letter.js
import { rest } from 'msw';

const date = new Date('2022-04-04T23:50:00');
const baseURL = 'http://timemachineletter.tk:8080/';
const dummy = {
  readable: true,
  id: 3,
  createAt: '2019-03-28',
  openAt: date,
  title: '제목1',
  content: '첫번째 테스트 본문입니다',
  letterFrom: '개발자',
  letterTo: '뭐',
};
export const Letter = [
  rest.get(`${baseURL}letter/:hash`, (req, res, ctx) => res(ctx.status(200), ctx.json(dummy))),
  rest.delete(`${baseURL}letter/:hash`, (req, res, ctx) => res(ctx.status(200))),
];
